### PR TITLE
Interactive list

### PR DESCRIPTION
**Pull request by over-geek**
## TODO-List: Add-Remove-Edit Functionalities

In this milestone, I:
- Implemented a function for for the "Clear all completed" button (use filter() method).
- Updated items object's value for completed key upon user actions.

Hi Code Reviewer, for the past day I've been getting linter webhint error preventing my project from being reviewed. Please I request that you review my project with the linter webhint error. Thank you

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import './style.css';
 import addTask from './modules/addTask.js';
-import { addTaskItem, getTask, updateTask } from './modules/storeTask.js';
+import {
+  addTaskItem, deleteTask, getTask, updateTask,
+} from './modules/storeTask.js';
+import handleCompleted from './modules/handleCompleted.js';
 import refresh from './assets/refresh.png';
 import enter from './assets/enter.png';
+import removeTask from './modules/removeTask';
```
Comment (abdulhamiid) [inline src/index.js:9]: Please kindly fix this `import` error in `src/index.js`
image

Expected 🤓✔️ 
```JavaScript
import removeTask from './modules/removeTask.js';
```

File: src/modules/handleCompleted.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { getTask } from './storeTask.js';
+
+const handleCompleted = (index) => {
+  const completed = document.getElementById(`check${index}`).toggleAttribute('checked');
+  const tasks = getTask();
+
+  tasks[index].completed = completed;
+  localStorage.setItem('data', JSON.stringify(tasks));
+};
```
Comment (abdulhamiid) [inline src/modules/handleCompleted.js:9]: According to the project requirements, we should update the task object's values for completed `Items` upon user actions.

However, In your app, when I marked `Item One` and `Item Two` as completed, different items were updated 👀, therefore that is affecting the result of the "Clear all completed" button.

image

Kindly fix this issue. 🛠😊

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @over-geek 👋, 
Alt Text
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️Use webpack to bundle JavaScript.
- ✔️ Learn how to use proper ES6 syntax.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tugsem — CHANGES_REQUESTED** [review]
## STATUS: REQUIRED CHANGES ♻️
Hello @over-geek 👋 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text

 ## Highlights 🚀
✔️Good job on solving the issue reported by the previous reviewer

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by over-geek** [comment]
@tugsem Please what changes?

**Review by joskalenda — CHANGES_REQUESTED** [review]
Hi @over-geek , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ❌

> You have submitted a project, that has some failed linters, please fix them and resubmit your word






Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

### Cheers and Happy coding!👏👏👏

------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

-----
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._ 🟢

**Comment by over-geek** [comment]
@joskalenda Thank you for your review. I've been having a webhint error since yesterday and the solution hasn't yet been found. I requested that my project be reviewed with the linters error which has been accepted for all previous reviews as indicated in the pull request description.

**Review by Lameck1 — COMMENTED** [review]
Hello @over-geek 👋,

_This is an additional review as a result of the second opinion request._

I am sorry for what has transpired. It's true that @tugsem did not highlight what changes you need to make. However, @tugsem has once again claimed your newest code review request.

@tugsem, let's hope that you are going to review this PR and leave desirable feedback. Thank you.

Keep rocking 🚀!

**Comment by tugsem** [comment]
> Hello @over-geek 👋,
> 
> _This is an additional review as a result of the second opinion request._
> 
> I am sorry for what has transpired. It's true that @tugsem did not highlight what changes you need to make. However, @tugsem has once again claimed your newest code review request.
> 
> @tugsem, let's hope that you are going to review this PR and leave desirable feedback. Thank you.
> 
> Keep rocking 🚀!

@Lameck1 I am assisting the student via Slack right now, when he applies the changes, I am going to approve it. Thank you

**Comment by tugsem** [comment]
## STATUS: APPROVED 👍🎯  
Hello @over-geek 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ All items have the right indexes
✔️deleting an item updates the indexes as expected
✔️ all checked items persist in the local storage
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**